module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -236,337 +331,382 @@ namespace module::input {
-                }
+                        // Filter is not reliable, just use previous sensors
+                        if (reset_filter.load()) {
+                            if (previousSensors) {
+                                sensors->Htw              = previousSensors->Htw;
+                                sensors->Hgt              = previousSensors->Hgt;
+                                sensors->Hgc              = previousSensors->Hgc;
+                                sensors->angular_position = previousSensors->angular_position;
+                            }
+                        }
+                        else {
+                            // Gives us the quaternion representation
+                            const auto o = MotionModel<double>::StateVec(motionFilter.get());
+
+                            // Map from world to torso coordinates (Rtw)
+                            Eigen::Affine3d Hwt;
+                            Hwt.linear()      = o.Rwt.toRotationMatrix();
+                            Hwt.translation() = o.rTWw;
+                            sensors->Htw      = Hwt.inverse().matrix();
+
+                            // Integrate gyro to get angular positions
+                            sensors->angular_position = o.omegaTTt / 90.0;
+
+                            if (this->config.debug) {
+                                log("p_x:",
+                                    sensors->angular_position.x(),
+                                    "p_y:",
+                                    sensors->angular_position.y(),
+                                    "p_z:",
+                                    sensors->angular_position.z());
+                            }
+
+                            /************************************************
+                             *                  Kinematics Horizon          *
+                             ************************************************/
+                            Eigen::Affine3d Rwt(sensors->Htw.inverse());
+                            // remove translation components from the transform
+                            Rwt.translation() = Eigen::Vector3d::Zero();
+                            Eigen::Affine3d Rgt(
+                                Eigen::AngleAxisd(-Rwt.rotation().eulerAngles(0, 1, 2).z(), Eigen::Vector3d::UnitZ())
+                                * Rwt);
+                            // sensors->Hgt : Mat size [4x4] (default identity)
+                            // createRotationZ : Mat size [3x3]
+                            // Rwt : Mat size [3x3]
+                            sensors->Hgt = Rgt.matrix();
+                            auto Htc     = sensors->Htx[ServoID::HEAD_PITCH];
+
+                            // Get torso to world transform
+                            Eigen::Affine3d yawlessWorldInvR(
+                                Eigen::AngleAxisd(-Hwt.rotation().eulerAngles(0, 1, 2).z(), Eigen::Vector3d::UnitZ())
```

Comment:
needs comment. I know what this does, but it is certainly not intuitive

Reply:
This code pre-dates this PR. Line is only modified because of whitespace changes

Reply:
hmmm, ok. sure